vsdkx/addon/tracking/speed_estimation_processor.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+
+from vsdkx.core.interfaces import Addon
+from vsdkx.core.structs import AddonObject
+
+
+class SpeedEstimationProcessor(Addon):
+    """
+    Calculate movement direction for objects based on their past and present
+    coordinates on the frame. This algorithm is based on the methodology
+    proposed on the following paper:
+
+    https://ieeexplore.ieee.org/document/6614066
+
+    Attributes:
+        person_action: (boolean) Flag on estimating actions such as
+            standing | walking | running
+        camera_length: (int|float) Distance from camera to furthest
+            object/ landmark in meters
+        fps: (int) Input frame rate
+        lens_dimension: (int) Camera lens dimension in millimeters. Optional
+            value if 'camera_horizontal_degrees' and
+            'camera_vertical_degrees' are specified instead.
+        focal_length: (int) Camera's focal length in millimeters. Optional
+            value if 'camera_horizontal_degrees' and
+            'camera_vertical_degrees' are specified instead.
+        camera_horizontal_degrees: (int) Camera's horizontal view in degrees.
+            Optional value if 'lens_dimension' and 'focal_length'
+            are specified instead.
+        camera_vertical_degrees: (int) Camera's vertical view in degrees.
+            Optional value if 'lens_dimension' and 'focal_length'
+            are specified instead.
+        _walking_action: (string) Constant string for walking action
+        _running_action: (string) Constant string for running action
+        _standing_action: (string) Constant string for standing action
+        _kph: (float) Constant for kilometers per hour (kph)
```

Comment:
can you explain this better please? it's a speed constant, but for what?

Reply:
Updated